src/value_wire.rs

```diff
@@ -235,6 +235,21 @@ impl ValueWire {
         tree_sum(&sum_terms, &a.id_gen)
     }
 
+    pub fn exp(a: &ValueWire, b: &ValueWire) -> ValueWire {
+        match b.as_usize() {
+            Some(n) => {
+                let mut e = a.clone();
+
+                for _i in 0..n - 1 {
+                    e = ValueWire::mul(&e, &e);
+                }
```

Comment:
I know this is just a draft, but don't forget exponentiation by squaring

```
eg x^8
  = x * x * x * x * x * x * x * x // 7 multiplications
  = ((x^2)^2)^2 // 3 multiplications

eg x^13
  = x * x * x * x * x * x * x * x * x * x * x * x * x // 12 multiplications
  = x * (x^2)^2 * ((x^2)^2)^2 // 5 multiplications
```

https://en.wikipedia.org/wiki/Exponentiation_by_squaring